Source/WebCore/page/Frame.cpp

```diff
@@ -104,4 +105,12 @@ void Frame::disconnectOwnerElement()
         document->frameWasDisconnectedFromOwner();
 }
 
+void Frame::takeWindowProxy(Frame& frame)
```

Comment:
I find the naming slightly confusing. Maybe it should be `takeWindowProxyFrom()`.